src/metadata/selectors.js

```diff
@@ -85,6 +85,29 @@ export const getCategoriesByCategoryComboId = createCachedSelector(
         categoryCombo.categories.map((id) => categories[id])
 )((_, categoryComboId) => categoryComboId)
 
+/**
+ * @param {*} metadata
+ * @param {*} dataSetId
+ */
+export const getCategoryComboByDataSetId = createCachedSelector(
+    getDataSetById,
+    getCategoryCombos,
+    (dataSet, categoryCombos) => {
+        return categoryCombos?.[dataSet.categoryCombo.id]
+    }
+)((_, categoryComboId) => categoryComboId)
```

Comment:
Note that the `categoryComboId` variable is actually `dataSetId`. It has the same signature as the function - and generated a key based on that. This works though, but confusing name.